app/src/main/java/org/schabi/newpipe/database/feed/dao/FeedDAO.kt

```diff
@@ -0,0 +1,150 @@
+package org.schabi.newpipe.database.feed.dao
+
+import androidx.room.*
+import io.reactivex.Flowable
+import org.schabi.newpipe.database.feed.model.FeedEntity
+import org.schabi.newpipe.database.feed.model.FeedLastUpdatedEntity
+import org.schabi.newpipe.database.stream.model.StreamEntity
+import org.schabi.newpipe.database.subscription.SubscriptionEntity
+import java.util.*
+
+@Dao
+abstract class FeedDAO {
+    @Query("DELETE FROM feed")
+    abstract fun deleteAll(): Int
+
+    @Query("""
+        SELECT s.* FROM streams s
+
+        INNER JOIN feed f
+        ON s.uid = f.stream_id
+
+        ORDER BY s.upload_date IS NULL DESC, s.upload_date DESC, s.uploader ASC
+
+        LIMIT 500
```

Comment:
Paged lists were not used in this PR, so I felt that a hard limit were needed. But that is just an arbitrary number that seemed reasonable at the time.

So yeah, this should be fixed when paged lists are used.

> This might be too high and cause freezes on old phones

As the list are composed by essentially a bunch of POJOs, I doubt it. And remember that we use RecyclerView, it only holds some views required to fill the screen + some wiggle room.

Reply:
Good. In case people experience problems, we can still change this.